src/prefect/orion/api/server.py

```diff
@@ -267,27 +267,35 @@ async def wrapper(*args, **kwargs):
         )
 
         memo_store_path = PREFECT_MEMO_STORE_PATH.value()
-        if memo_store_path.exists():
-            saved_blocks_loading_hash = toml.load(memo_store_path).get(
-                "block_auto_registration"
+        try:
+            if memo_store_path.exists():
+                saved_blocks_loading_hash = toml.load(memo_store_path).get(
+                    "block_auto_registration"
+                )
+                if (
+                    saved_blocks_loading_hash is not None
+                    and current_blocks_loading_hash == saved_blocks_loading_hash
+                ):
+                    if PREFECT_DEBUG_MODE.value():
+                        logger.debug(
+                            "Skipping block loading due to matching hash for block "
+                            "auto-registration found in memo store"
+                        )
+                    return
+        except Exception as exc:
+            logger.warn(
+                f"Unable to read memo_store.toml during block auto-registration: {exc!r}"
```

Comment:
The full path that you attempted to write to would be good. Additionally, a sentence explaining the implication of not being able to write to this file.

Reply:
Updated the messages in [d4823f5](https://github.com/PrefectHQ/prefect/pull/6997/commits/d4823f5cc694f895e23df368716d2ba004e98175). Let me know what you think!